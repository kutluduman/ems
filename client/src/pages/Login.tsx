import { Link as RouterLink } from "react-router-dom";
import { styled } from "@mui/material/styles";
import { Card, Stack, Link, Container, Typography } from "@mui/material";

import AuthLayout from "../layouts/AuthLayout";
import Page from "../components/Page";
import { MHidden, Width } from "../components/@material-extend";
import LoginForm from "../components/authentication/LoginForm";

const RootStyle = styled(Page)(({ theme }) => ({
  [theme.breakpoints.up("md")]: {
    display: "flex",
  },
}));

const SectionStyle = styled(Card)(({ theme }) => ({
  width: "100%",
  maxWidth: 464,
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  margin: theme.spacing(2, 0, 2, 2),
}));

const ContentStyle = styled("div")(({ theme }) => ({
  maxWidth: 480,
  margin: "auto",
  display: "flex",
  minHeight: "100vh",
  flexDirection: "column",
  justifyContent: "center",
  padding: theme.spacing(12, 0),
}));

const Login = () => {
  return (
    <RootStyle title="Login | Minimal-UI">
      <AuthLayout>
        Don’t have an account? &nbsp;
        <Link
          underline="none"
          variant="subtitle2"
          component={RouterLink}
          to="/register"
        >
          Get started
        </Link>
      </AuthLayout>

      <MHidden width={Width.mdDown}>
        <SectionStyle>
          <Typography variant="h3" sx={{ px: 5, mt: 10, mb: 5 }}>
            Hi, Welcome Back
          </Typography>
          <img
            src="https://minimal-kit-react.vercel.app/static/illustrations/illustration_login.png"
            alt="login"
          />
        </SectionStyle>
      </MHidden>

      <Container maxWidth="sm">
        <ContentStyle>
          <Stack sx={{ mb: 5 }}>
            <Typography variant="h4" gutterBottom>
              Sign in to EMS
            </Typography>
            <Typography sx={{ color: "text.secondary" }}>
              Enter your details below.
            </Typography>
          </Stack>

          <LoginForm />

          <MHidden width={Width.smUp}>
            <Typography variant="body2" align="center" sx={{ mt: 3 }}>
              Don’t have an account?&nbsp;
              <Link variant="subtitle2" component={RouterLink} to="register">
                Get started
              </Link>
            </Typography>
          </MHidden>
        </ContentStyle>
      </Container>
    </RootStyle>
  );
}

export default Login;
