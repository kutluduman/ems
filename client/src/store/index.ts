import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;