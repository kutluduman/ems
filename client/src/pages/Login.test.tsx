import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import ThemeConfig from "../theme";
import { store } from "../store";
import Login from "./Login";

describe("Login Component", () => {
  test("renders the Login Component", () => {
    const tree = create(
      <Provider store={store}>
        <ThemeConfig>
          <Router>
            <Login />
          </Router>
        </ThemeConfig>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});