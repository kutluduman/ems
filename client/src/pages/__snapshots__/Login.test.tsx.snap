// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders the Login Component 1`] = `
<div
  className="css-v82cpl"
>
  <header
    className="css-103o8ot"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="MuiBox-root css-1ngw63v"
        src="logo.svg"
      />
    </a>
    <p
      className="MuiTypography-root MuiTypography-body2 css-p34ji4-MuiTypography-root"
    >
      Don’t have an account?  
      <a
        className="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineNone css-1ser5cc-MuiTypography-root-MuiLink-root"
        href="/register"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Get started
      </a>
    </p>
  </header>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      className="css-26q4qm"
    >
      <div
        className="css-lr1r4n-MuiStack-root"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
        >
          Sign in to EMS
        </h4>
        <p
          className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Enter your details below.
        </p>
      </div>
      <form
        action="#"
        autoComplete="off"
        noValidate={true}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="css-1lwbda4-MuiStack-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              Email address
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-173wfxe"
                >
                  <span>
                    Email address
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
            >
              Password
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="current-password"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-173wfxe"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="css-1ulybn8-MuiStack-root"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                name="rememberMe"
                onChange={[Function]}
                type="checkbox"
                value={true}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Remember me
            </span>
          </label>
        </div>
        <button
          className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Login
        </button>
      </form>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1erjnf3-MuiTypography-root"
      >
        Don’t have an account? 
        <a
          className="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineAlways css-id1q2y-MuiTypography-root-MuiLink-root"
          href="/register"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Get started
        </a>
      </p>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mmxaym-MuiPaper-root-MuiCard-root"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-1f32e3x-MuiTypography-root"
    >
      Welcome Back!
    </h3>
    <img
      alt="login"
      className="css-14z6jlm"
      src="login.svg"
    />
  </div>
</div>
`;
